portfolio/src/main/java/com/google/sps/data/FindSchedule.java

```diff
@@ -0,0 +1,116 @@
+
+    // This will iterate through the time ranges and tasks and if one can be
+    // scheduled then it will be (and we move onto the next task) otherwise
+    // we move on to the next range (this is because the tasks are sorted by
+    // duration so if one task did not fit in the given range then we know no
+    // later ones will fit either). We create new Task objects for the result
+    // so data structures passed in are never changed.
+    while (i < availableTimes.length && j < tasksArray.length) {
+      TimeRange timeRange = availableTimes[i];
+      Task task = tasksArray[j];
+      // Either time is already past the start of the time range or we should
+      // update it (maybe this is our first iteration in the range).
+      scheduleTimeSeconds = Math.max(scheduleTimeSeconds, timeRange.start());
+      // The task can be scheduled in the current time range.
+      if (scheduleTimeSeconds + task.getDurationSeconds() <= timeRange.end()) {
+        Task scheduledTask;
+        if (task.getDescription().isPresent()) {
+          scheduledTask =
+              new Task(task.getName(), task.getDescription().get(), task.getDuration().toMinutes(),
+                  task.getPriority(), Instant.ofEpochSecond(scheduleTimeSeconds).toString());
+        } else {
+          scheduledTask = new Task(task.getName(), null, task.getDuration().toMinutes(),
+              task.getPriority(), Instant.ofEpochSecond(scheduleTimeSeconds).toString());
+        }
+
+        scheduledTasks.add(scheduledTask);
+        scheduleTimeSeconds += task.getDurationSeconds();
+        j++;
+      } else {
+        i++;
+      }
+    }
+    return scheduledTasks;
+  }
+
+  /**
+   * This method returns an TimeRange array which represent the periods of
+   * time that are empty of events and lie completely inside the person's
+   * working hours.
+   */
+  public static TimeRange[] emptyTimeRanges(
+      List<CalendarEvent> events, long workHoursStartTimeSeconds, long workHoursEndTimeSeconds) {
+    List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    // This represents the earliest time that we can schedule a window for the
+    // meeting. As events are processed, this changes to their end times.
+    long earliestPossibleSoFar = workHoursStartTimeSeconds;
+    for (CalendarEvent event : events) {
+      // Make sure that there is some time between the events and it is not
+      // later than the person's working hours ending time.
+      if (event.getStartTimeLong() - earliestPossibleSoFar > (long) 0
+          && event.getStartTimeLong() <= workHoursEndTimeSeconds) {
+        possibleTimes.add(TimeRange.fromStartEnd(
+            earliestPossibleSoFar, event.getStartTimeLong(), /* inclusive= */ true));
+      }
+      earliestPossibleSoFar = Math.max(earliestPossibleSoFar, event.getEndTimeLong());
+    }
+    // The end of the work hours is potentially never included so we check.
+    if (workHoursEndTimeSeconds - earliestPossibleSoFar > 0) {
```

Comment:
Equivalent to,

workHoursEndTimeSeconds > earliestPossibleSoFar

Reply:
Done.